### ISSUE TITLE
caret moves review cursor not working after using Win +V

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#13812]
### Steps to reproduce:
1. Press `nvda+6` to turn caret moves review cursor on;
2. Enable clipboard history in Windows 10 or windows11;
3. Open Windows Notepad;
4. Type 'ABCD' in notepad;
5. `ctrl+A, ctrl+X` cut to clipboard;
6. `Win+V` to open the clipboard history and paste the 'ABCD' you just copied;
7. Try changing the caret position with the `left arrow`/`right arrow`.
8. Press `numpad2` to observe the review cursor position.
### Actual behavior:
The review cursor position is not consistent with the caret position:
For example, the caret position is after 'D', but pressing `numpad2` NVDA reports 'A';
### Expected behavior:
The review cursor position should be synchronized with the caret position.
#### NVDA installed/portable/running from source:
installed and portable
#### NVDA version:
2022.2Beta1
#### Windows version:
Windows 10 21H2 (x64) build 19044.1766
#### Name and version of other software in use when reproducing the issue:
Windows Clipboard History
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA2022.1 can also reproduce this
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by michaelDCurran** [comment]
So far I cannot reproduce this on Windows 11. Is it only a problem in Notepad? what other applications?

**Comment by cary-rowen** [comment]
@michaelDCurran 
Hi, I can also reproduce this on Windows10 using Word, Notepad++.

**Comment by amirmahdifard** [comment]
@seanbudd @michaelDCurran please dix this bug. thank you so much

**Comment by josephsl** [comment]
Hi,

This occurs in Windows 10, a potential cause being focus and caret movement not being detected. This is ultimately caused by focus movement issue when opening and closing clipboard history (I tried resolving a similar issue with emoji panel in Windows 10 a fe years ago to no avail). Specifically:

1. Open Notepad.
2. Type something, cut it, then open clipboard history.

At this point, clipboard history window opens but system focus is still located in Notepad. This is because emoji panel and clipboard history are overlay windows. Although a menu open event is fired, NVDA sees it as a window open event, and clipboard history does not move system focus to itself. At the same time, UIA element selected event is also fired in some cases, causing navigator object to move to the first selected item. After making a selection in clipboard history by pressing Enter, NVDA's idea of focused object is still Notepad, and navigator object gets stuck, or in this case, perhaps stay in the cursor location prior to opening clipboard history in the first place. This is resolved in Windows 11 where clipboard history does move system focus when opened and a selection is made.

If it turns out to be something going on with Windows 10 clipboard history, then I'm afraid it may not be resolved by Microsoft as Windows 10 (vibranium at this point) is in maintenance mode. Resolving this from NVDA side will require focus tracking and movement workarounds for modern keyboard features such as clipboard history and emoji panel (this was the route I tried a few years ago to resolve similar issues with modern keyboard to no avail).

Thanks.
